sonar-plugin-api/src/main/java/org/sonar/api/batch/bootstrap/ProjectReactor.java

```diff
@@ -26,8 +26,11 @@
 
 /**
  * Mutable project definitions that can be modified by {@link ProjectBuilder} extensions.
+ * 
+ * @deprecated since 6.5 plugins should no longer modify the project's structure
  * @since 2.9
  */
+@Deprecated
 @ScannerSide
 public class ProjectReactor implements ProjectKey {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)